contracts/core/tokens/RebalancingSetTokenV3.sol

```diff
@@ -0,0 +1,159 @@
+     * [2]rebalanceInterval:        Minimum amount of time between rebalances
+     * [3]rebalanceFailPeriod:      Time after auctionStart where something in the rebalance has gone wrong
+     * [4]lastRebalanceTimestamp:   Time of the last rebalance; Allows customized deployments
+     * [5]entryFee:                 Mint fee represented in a scaled decimal value (e.g. 100% = 1e18, 1% = 1e16)
+     *
+     * @param _addressConfig             List of configuration addresses
+     * @param _uintConfig                List of uint addresses
+     * @param _name                      The name of the new RebalancingSetTokenV2
+     * @param _symbol                    The symbol of the new RebalancingSetTokenV2
+     */
+    constructor(
+        address[8] memory _addressConfig,
+        uint256[6] memory _uintConfig,
+        string memory _name,
+        string memory _symbol
+    )
+        public
+        RebalancingSetTokenV2(
+            _addressConfig,
+            _uintConfig,
+            _name,
+            _symbol
+        )
+    {}
+
+    /*
+     * Overrides the RebalancingSetTokenV2 settleRebalance function.
+     *
+     * After a successful rebalance, the new Set is issued. 
+     * Full issuance functionality is now returned to set owners.
+     *
+     * Anyone can call this function.
+     */
+    function settleRebalance()
+        external
+    {
+        RebalancingSettlement.validateRebalancingSettlement();
+
+        uint256 issueQuantity = RebalancingSettlement.calculateSetIssueQuantity(nextSet);
+        uint256 newUnitShares = RebalancingSettlement.calculateNextSetNewUnitShares(issueQuantity);
+
+        // The unit shares must result in a quantity greater than the number of natural units outstanding
+        validateUnitShares(newUnitShares);
+
+        RebalancingSettlement.issueNextSet(issueQuantity);
+
+        liquidator.settleRebalance();
+
+        // Rebalance index is the current vs next rebalance
+        emit RebalanceSettled(
+            address(0),      // No longer used
+            0,               // No longer used
+            0,               // No longer used
+            rebalanceIndex,
+            issueQuantity,
+            newUnitShares
+        );
+
+        RebalancingSettlement.transitionToDefault(newUnitShares);
+    }
+    /*
+     * During the Default stage, the incentive / rebalance Fee can be triggered. This will
+     * retrieve the current inflation fee from the fee calulator and mint the according
+     * inflation to the feeRecipient. The unit shares is then adjusted based on the new
+     * supply.
+     *
+     * Anyone can call this function.
+     */
+    function actualizeFee()
+        external
+    {
+        IncentiveFee.validateFeeActualization();
+
+        // Calculates fees and mints Rebalancing Set to the feeRecipient, increasing supply
+        (uint256 feePercent, uint256 feeQuantity) = RebalancingSettlement.handleFees();
```

Comment:
We might have an issue here because IFeeCalculator `getFee` is a `view` function but now alters state in order to set high watermarks and timestamps.

Reply:
Ok i see. Let's make it non-view.